docs/configuration.md

```diff
@@ -1 +1,363 @@
+
+  ```yaml
+apiVersion: rhdh.redhat.com/v1alpha2
+kind: Backstage
+metadata:
+  name: mybackstage
+spec:
+  application:
+    appConfig:
+      configMaps:
+        - name: my-app-config
+    extraEnvs:
+      secrets:
+        - name: my-secrets
+  ```
+
+This Custom Resource defines a Backstage instance called **mybackstage** and also:
+  - Adds additional app-config stored in the **my-app-config** ConfigMap.
+  - Adds some extra environment variables stored (as key-value pairs) in the Secret called **my-secrets**.
+
+For API version **v1alpha2** (Operator version **0.3.x**), the Backstage CR Spec contains the following top-level elements:
+
+  * [application](#application-configuration)
+  * [deployment](#deployment-configuration)
+  * [database](#local-database-configuration)
+  * [rawRuntimeConfig](#raw-configuration)
+  
+  ### Application Configuration
+  
+  This section explains how the Backstage Application is configured inside the container.
+  
+  #### app-config
+  
+  As documented in the [Backstage documentation](https://backstage.io/docs/conf/writing/), the Backstage application is configured with one or more app-config files, which are merged from the first to the last. The Operator can contribute to the app-config list by mounting the ConfigMap defined in `default-config/app-config.yaml`, as specified in the [Admin Guide](admin.md). Additionally, it is possible to define an array of external, user-created ConfigMaps located in the same namespace as the Backstage CR using `spec.application.appConfig`.
+
+For example, consider the following ConfigMaps containing Backstage app-config configuration:
+
+  ```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cm1
+data:
+  app-config1.yaml: |
+    # Some fragment of app-config here
+  app-config2.yaml: |
+    # Another fragment of app-config here
+  ```
+
+These app-config files can be mounted to the Backstage container as follows:
+
+  ```yaml
+spec:
+  application:
+    appConfig:
+      mountPath: /my/path
+      configMaps:
+        - name: app-config1
+        - name: app-config2
+  ```
+
+In this example, additional app-configs are defined in the ConfigMaps called **app-config1** and **app-config2**, and they are mounted as YAML files to the `/my/path` directory in the Backstage container:
+
+  ```
+  /my/path/app-config1.yaml
+  /my/path/app-config2.yaml
+  ```
+
+The ConfigMap key/value defines the file name and content, and these app-configs will be applied in the same order as they are declared in the CR. Therefore, as per this example, two ConfigMaps should be created in the namespace as a prerequisite. Then the Operator will create the files `/my/path/app-config1.yaml` and `/my/path/app-config2.yaml`, adding them to the end of the Backstage command-line arguments as follows:
+
+  ```
+  -app-config /my/path/app-config1.yaml -app-config /my/path/app-config2.yaml
```

Comment:
As depicted in https://github.com/redhat-developer/rhdh-operator/blob/main/api/v1alpha2/backstage_types.go#L92-L94, there is a limitation when using a single ConfigMap containing several filenames: the order in which they will translate into the container args is not guaranteed.
I think we should call this out explicitly here too. 
And also maybe update your example to use 2 different ConfigMaps, each with a single filename? (that's what you are doing by the way in the CR example where you are specifying 2 ConfigMaps in `spec.application.appConfig.configMaps`)

Reply:
good point, fixing it